e2e/cypress/integration/notifications/desktop_notifications_spec.js

```diff
@@ -270,6 +270,174 @@ describe('Desktop notifications', () => {
             });
         });
     });
+
+    it('MM-T491 - Channel notifications: No desktop notification when in focus', () => {
+        cy.apiCreateUser().then(({user}) => {
+            cy.apiAddUserToTeam(testTeam.id, user.id);
+            cy.apiLogin(user);
+
+            cy.apiGetChannelByName(testTeam.name, 'Off-Topic').then(({channel}) => {
+                const message = '/echo test 3';
+
+                // # Go to Off topic
+                cy.visit(`/${testTeam.name}/channels/${channel.name}`);
+                spyNotificationAs('withNotification', 'granted');
+
+                // Have another user send a post with delay
+                cy.postMessageAs({sender: testUser, message, channelId: channel.id});
+
+                // * Desktop notification is not received
+                cy.get('@withNotification').should('not.have.been.called');
+            });
+        });
+    });
+
+    it('MM-T491 - Channel notifications: No desktop notification when in focus', () => {
+        cy.apiCreateUser().then(({user}) => {
+            cy.apiAddUserToTeam(testTeam.id, user.id);
+            cy.apiLogin(user);
+
+            cy.apiGetChannelByName(testTeam.name, 'Off-Topic').then(({channel}) => {
+                const message = '/echo test 3';
+
+                // # Go to Off topic
+                cy.visit(`/${testTeam.name}/channels/${channel.name}`);
+                spyNotificationAs('withNotification', 'granted');
+
+                // Have another user send a post with delay
+                cy.postMessageAs({sender: testUser, message, channelId: channel.id});
+
+                // * Desktop notification is not received
+                cy.get('@withNotification').should('not.have.been.called');
+            });
+        });
+    });
+
+    it('MM-T494 - Channel notifications: Send Desktop Notifications - Only mentions and DMs', () => {
+        cy.apiCreateUser().then(({user}) => {
+            cy.apiAddUserToTeam(testTeam.id, user.id);
+            cy.apiLogin(user);
```

Comment:
if you are going to create a user for the whole spec, I think it is better to create it in the `before` section and reuse. unless you need a different one each time.

Reply:
I agree with the comment above. Though it became a pattern on this spec file, it may worth changing properly so it will not be copied over in the future. (But I'm also good to remain as is and I'll handle the clean up later.)